db/schema_tables.cc

```diff
@@ -2691,14 +2691,32 @@ static void make_update_indices_mutations(
         make_drop_table_or_view_mutations(views(), view, timestamp, mutations);
     }
 
-    // newly added indices and old indices with updated attributes
-    for (auto&& name : boost::range::join(diff.entries_differing, diff.entries_only_on_right)) {
+    // old indices with updated attributes
+    for (auto&& name : diff.entries_differing) {
+        const index_metadata& index = new_table->all_indices().at(name);
+        add_index_to_schema_mutation(new_table, index, timestamp, indices_mutation);
+        auto& cf = db.find_column_family(new_table);
+        auto view = cf.get_index_manager().create_view_for_index(index, new_token_column_computation);
+        auto view_mutations = make_view_mutations(view, timestamp, true);
+        view_mutations.copy_to(mutations);
+    }
+    // Newly added indices. Because these are newly created tables (views),
+    // we need to call the before_create_column_family callback for them.
+    // If we don't, among other things *tablets* will not be created for
+    // these new views.
+    // The callbacks must be called in a Seastar thread, which means that
+    // *this* function must be called in a Seastar thread when creating an
+    // index.
+    for (auto&& name : diff.entries_only_on_right) {
         const index_metadata& index = new_table->all_indices().at(name);
         add_index_to_schema_mutation(new_table, index, timestamp, indices_mutation);
         auto& cf = db.find_column_family(new_table);
         auto view = cf.get_index_manager().create_view_for_index(index, new_token_column_computation);
         auto view_mutations = make_view_mutations(view, timestamp, true);
         view_mutations.copy_to(mutations);
```

Comment:
Maybe worth extracting this to a function/lambda to avoid duplication?

Reply:
Maybe, I'll take a look how to do that in the least ugly way. Although the duplicates are just a few lines from each other so it's hard to make a mistake.

Reply:
I'll do it with a lambda, and send a new version of this PR in a few minutes.